src/workerd/api/crypto-impl-asymmetric.c++

```diff
@@ -103,6 +103,59 @@ public:
         "Asymmetric signing requires a private key.");
 
     auto type = lookupDigestAlgorithm(chooseHash(algorithm.hash)).second;
+    if (getAlgorithmName() == "RSASSA-PKCS1-v1_5") {
+      // RSASSA-PKCS1-v1_5 requires the RSA key to be at least as big as the digest size
+      // plus a 15 to 19 byte digest-specific prefix (see boringssl's RSA_add_pkcs1_prefix) plus 11
+      // bytes for padding (see RSA_PKCS1_PADDING_SIZE). For simplicity, require the key to be at
+      // least 32 bytes larger than the hash digest.
+      // Similar checks could also be adopted for more detailed error handling in verify(), but the
+      // current approach should be sufficient to avoid internal errors.
+      RSA& rsa = JSG_REQUIRE_NONNULL(EVP_PKEY_get0_RSA(getEvpPkey()), DOMDataError,
```

Comment:
The only remaining question here is whether it's possible that adding the additional errors here could break any existing code. If so, these *might* need to go behind a compatibility flag. If these are cases that simply wouldn't have worked correctly previously (causing an internal error as opposed to a useful DOMException) then cool, no compat flag needed. If, however, these cases previously would not have thrown, adding the error could break someone.

Reply:
Overall, I'm confident that after switching to just logging warnings in some cases, the additional error statements will only produce errors when there previously would have been an internal error/a different error message. In this function the math gets relatively complex but based on boringssl's code and my tests it should be correct.